Build failure: corkscrew

Corkscrew is broken for `aarch64-darwin` on both `23.11` and unstable

### Steps To Reproduce

`nix build nixpkgs/nixos-23.11#corkscrew` and `nix build nixpkgs-unstable#corkscrew` fail with the same error (see below for build log).

### Build log

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/5yfmrs1b6s2pg9612s06jxy9klcj7m92-corkscrew-2.0.tar.gz
source root is corkscrew-2.0
setting SOURCE_DATE_EPOCH to timestamp 998591634 of file corkscrew-2.0/README
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
configure flags: --prefix=/nix/store/2jylazj7dl6s089155y9b15qlqq26jck-corkscrew-2.0
creating cache ./config.cache
checking for a BSD compatible install... /nix/store/h0fgpyxfav7ybnw5sdg8jpyv9f95x4w0-coreutils-9.3/bin/install -c
checking whether build environment is sane... yes
checking whether make sets ${MAKE}... yes
checking for working aclocal... missing
checking for working autoconf... missing
checking for working automake... missing
checking for working autoheader... missing
checking for working makeinfo... missing
checking for gcc... clang
checking whether the C compiler (clang  ) works... no
configure: error: installation or configuration problem: C compiler cannot create executables.
```

### Notify maintainers

No maintainers listed

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
λ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.6.0, macOS 13.5.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.10.3`
 - nixpkgs: `/nix/store/mpsjbqbyzcdjbgf2phj0yik4nk8dp1xv-mbz4hixfgxq5b6vc0k3pp2iglcd4c353-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/networking/corkscrew/default.nix
+++ b/pkgs/tools/networking/corkscrew/default.nix
@@ -1,23 +1,22 @@
-{ lib, stdenv, fetchurl, automake }:
+{ lib, stdenv, fetchFromGitHub, autoreconfHook }:
 
 stdenv.mkDerivation rec {
   pname = "corkscrew";
   version = "2.0";
-
-  src = fetchurl {
-    url = "http://agroman.net/corkscrew/corkscrew-${version}.tar.gz";
-    sha256 = "0d0fcbb41cba4a81c4ab494459472086f377f9edb78a2e2238ed19b58956b0be";
+  src = fetchFromGitHub {
+    owner = "bryanpkc";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-JiddvTbuOysenrVWGUEyKSzpCF1PJaYWQUdz3FuLCdw=";
   };
 
-  preConfigure = ''
-    ln -sf ${automake}/share/automake-*/config.sub config.sub
-    ln -sf ${automake}/share/automake-*/config.guess config.guess
-  '';
+  nativeBuildInputs = [ autoreconfHook ];
 
   meta = with lib; {
-    homepage    = "http://agroman.net/corkscrew/";
+    homepage = "https://github.com/bryanpkc/corkscrew";
     description = "A tool for tunneling SSH through HTTP proxies";
     license = lib.licenses.gpl2;
     platforms = platforms.unix;
   };
 }
+

